src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -5779,7 +5789,14 @@ private void AppendKeywords(StringBuilder sb, ulong keywords, string eventName)
             }
         }
 
-        private string GetTypeName(Type type)
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2072:UnrecognizedReflectionPattern",
+        Justification = "The recursive call GetType is safe since the type is not an enum in " +
+        "the 2nd call")]
+        private string GetTypeName(
+#if !ES_BUILD_STANDALONE
+            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicFields | DynamicallyAccessedMemberTypes.NonPublicFields)]
```

Comment:
Can we just suppress the warning for `type.GetFields` here? I don't think the ILLinker can ever remove the `value__` field from an Enum.

Reply:
Can we just make this call `Type.GetEnumUnderlyingType`?

EventSource is basically inlining what the default implementation of this is:

https://github.com/dotnet/runtime/blob/867ba801d1a609f4cbb6e5847a3ec030adb2d8f4/src/libraries/System.Private.CoreLib/src/System/Type.cs#L485-L495

And it skips any possible optimizations the RuntimeType might have.

https://github.com/dotnet/runtime/blob/867ba801d1a609f4cbb6e5847a3ec030adb2d8f4/src/libraries/System.Private.CoreLib/src/System/RuntimeType.cs#L145-L151

It's odd to have odd code that we then need to create suppressions for.